homeassistant/components/camera/__init__.py

```diff
@@ -614,6 +673,36 @@ def async_update_token(self) -> None:
             hashlib.sha256(_RND.getrandbits(256).to_bytes(32, "little")).hexdigest()
         )
 
+    async def async_added_to_hass(self) -> None:
```

Comment:
@bdraco I would like to get your opinion on if there is a better way to accomplish this check.

Effectively we want to update a member variable that is input into `state_attributes` any time a new provider is registered (see call above) or ideally any time we have a hunch the source may change.  (Technically, the source can change on ever call, but i think its OK to not worry about that).  

Right now this is called on startup only, which may be ok. But the other question is: integrations need to call super() on the parent to invoke this, or also it might send an RPC and fail.

Would love to hear your suggestions if you have any improvements.

Reply:
You could implement this in `async_internal_added_to_hass` to ensure it always happens.   Since this is base platform, I think this would be ok to do as long as you are careful to call the `super()` method

https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/entity.py#L799
https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/entity.py#L830

Reply:
Thanks, I saw this but was on the fence about using that. I agree it seems fine since this is a base platform. Updated.